react/features/toolbox/components/AbstractToolboxItem.js

```diff
@@ -0,0 +1,192 @@
+// @flow
+
+import { Component } from 'react';
+
+type Styles = {
+
+    /**
+     * Style for the item's icon.
+     */
+    iconStyle: Object,
+
+    /**
+     * Style for the item itself.
+     */
+    style: Object,
+
+    /**
+     * Color for the item underlay (shows when clicked).
+     */
+    underlayColor: string
+};
+
+export type Props = {
+
+    /**
+     * A succinct description of what the item does. Used by accessibility
+     * tools and torture tests.
+     */
+    accessibilityLabel: string,
+
+    /**
+     * Whether this item is disabled or not. When disabled, clicking an the item
+     * has no effect, and it may reflect on its style.
+     */
+    disabled: boolean,
+
+    /**
+     * The name of the icon of this {@code ToolboxItem}.
+     */
+    iconName: string,
+
+    /**
+     * The text associated with this item. When `showLabel` is set to
+     * {@code true}, it will be displayed alongside the icon.
+     */
+    label: string,
+
+    /**
+     * On click handler.
+     */
+    onClick: Function,
+
+    /**
+     * Whether to show the label or not.
+     */
+    showLabel: boolean,
+
+    /**
+     * Collection of styles for the item. Used only on native.
```

Comment:
I'm not super-thrilled about taking all props here, but after trying a number of things I couldn't come up with a way to handle optional i18n which would share all the code across web and native.